wip-stringparsexml.md

```diff
@@ -0,0 +1,168 @@
+<pre>
+  WIP: WIP-0021
+  Layer: Consensus (hard fork)
+  Title: Add StringParseXMLMap to RADON operators
+  Authors: Luis Rubio <luisr@witnet.foundation>
+  Discussions-To: `#dev-general` channel on Witnet Community's Discord server
+  Status: Draft
+  Type: Standards Track
+  Created: 2021-12-14
+  License: BSD-2-Clause
+</pre>
+
+
+## Abstract
+
+The `StringParseXMLMap` operator is proposed to be included in the operators catalog available in the RAD engine. Data requests
+will then support XML parsing as part of the **retrieval** scripts.
+
+
+## Motivation and rationale
+
+In order to increase the flexibility and potential use cases of the Witnet protocol, it is paramount to continue 
+expanding the catalog of available operators.
+
+This document proposes adding support for **XML parsing**, thus allowing the protocol to extract information from XML APIs
+or XML-encapsulated strings to be retrieved through data requests.
+
+## Specification
+
+The `StringParseXMLMap` operator converts a `String` (which is assumed to represent a valid XML document) into a `Map`, 
+following the next rules:
+- Every element MUST be turned into a `Map` in which the tag name will be inserted as the key, **attributes** 
+  would use the *@* symbol to precede their key name, and the elements inside would be also turned in another `Map`.
+  
+```xml
+<tag1 attr1="001">
+    <in_tag1>[...]</in_tag1>
+    <in_tag2>[...]</in_tag2>
+</tag1>
+```
+```rust
+{
+    tag1: {
+        @attr1: "001",
+        in_tag1: {...},
+        in_tag2: {...},
+    }
+}
+```
+
+- Leaf elements that have no attributes or children other than textual content MUST be added as an entry in the `Map` of
+  their parent element using their tag as the keys and their textual content as the values.
+
+```xml
+<tag1 attr1="001">
+    <in_tag1>Hello</in_tag1>
+    <in_tag2>[...]</in_tag2>
+</tag1>
+```
+```rust
+{
+    tag1: {
+        @attr1: "001",
+        in_tag1: "Hello",
+        in_tag2: {...},
+    }
+}
+```
+
+- Elements with the same tag name would be added as a `RadonArray` with the tag name in common as a *key* of the `Map`.
+
+```xml
+<tag1 attr1="001">
+    <in_tag>[...]</in_tag>
+    <in_tag>[...]</in_tag>
+</tag1>
+```
+```rust
+{
+    tag1: {
+        @attr1: "001",
+        in_tag: [
+            {...},
+            {...},
```

Comment:
Please use specific examples so that it is clear how the elements are collected into the array.